### PR TITLE
backport HLT developments from 9.2.x

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

File: DataFormats/EgammaReco/interface/ElectronSeed.h [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
-
-/** \class reco::ElectronSeed
- *
- * ElectronSeed is a seed for gsf tracking,  constructed from
- * either a supercluster or a ctf track.
- *
- * \author D.Chamont, U.Berthon, C.Charlot, LLR Palaiseau
- *
- ************************************************************/
+#ifndef DataFormats_EgammaReco_ElectronSeed_h
```
Comment (slava77) [inline DataFormats/EgammaReco/interface/ElectronSeed.h]: this file is probably the only change that triggers long recompilation.

Comment (Sam-Harper) [inline DataFormats/EgammaReco/interface/ElectronSeed.h]: yes it is. 

Unrelated to this there is a lot of other phaseI stuff needed to be compiled (its not just this E/gamma change which creates the situation David was complaining about).

**Comment by davidlange6** [comment]
so maybe we should see the full request before discussing further?

**Comment by fwyzard** [comment]
If I understood correctly, this _is_ the full request - it just is not only about E/Gamma.

I can change the subject if you prefer, of course.

**Comment by Sam-Harper** [comment]
Ah yes, there is more than just e/gamma here (I got confused from the title). I dont know what else is needed so wont comment on that. This has the e/gamma changes

**Comment by slava77** [comment]
@fwyzard 
a more complete title would be better so that it covers changes in this PR, at least with a broad brush

**Comment by fwyzard** [comment]
renamed.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by slava77** [comment]
@cmsbuild please test

@smuzaffar 
.. do we even have a test job running in bot for this?

**Comment by smuzaffar** [comment]
@slava77 , yes we have cms-bot test jobs for all active IBs (and 90X is still an active IB).

**Comment by smuzaffar** [comment]
please test with #20302

**Comment by slava77** [comment]
What was the baseline in 92X for the backport and was it actually a backport (as parts of the diffs look like new developments)?

I notice that the following do not exist in 92X or master and appear to be new developments that should be in master first
- SimMuon/MCTruth/plugins/HLTFilterToTrackProducer in entirety
- additions in Validation/RecoMuon/plugins/MuonTrackValidator 
- change in Validation/RecoMuon/python/MuonTrackValidator_cfi.py in ```maxPt```
- changes in RecoMuon/TrackerSeedGenerator/plugins/TSGForOI.cc

Then more:
- Changes in MuonTrackingRegionBuilder.cc, h and related revert the changes present in 90X already as of #18286 to some older version
- changes in RecoMuon/TrackerSeedGenerator/plugins/TSGForOI.cc and related appear to be incomplete: important bug fixes from #19947 are missing

**Comment by slava77** [comment]
@fwyzard @Sam-Harper 
please clarify on the origin of the additional changes as mentioned in https://github.com/cms-sw/cmssw/pull/20235#issuecomment-326716198
Thank you.

**Comment by Sam-Harper** [comment]
@slava77 , sorry, no idea. I took the latest 92X /93X for the electron stuff but thats not what you're asking out.

**Comment by slava77** [comment]
@fwyzard 
please clarify the status of this PR.

**Comment by fwyzard** [comment]
This is what was requested to further the HLT studies for Phase 2, based on the reports from @Sam-Harper @dsperka @battibass .
Personally, I cannot add anything nor comment their request.

If necessary, @gennai and marco.pieri@cern.ch can follow this up and comment further.

**Comment by gennai** [comment]
Apart from repeating what has already written by Andrea (i.e. these are backporting needed for upgrade studies) I cannot say much more.
About the technical comment on the missing code in the master, 
I have pinged on skype @battibass who promised to answer asap.

**Comment by gennai** [comment]
Apart from repeating what has already written by Andrea (i.e. these are backporting needed for upgrade studies) I cannot say much more.
About the technical comment on the missing code in the master, 
I have pinged on skype @battibass who promised to answer asap.

**Comment by battibass** [comment]
Hi all, 
firstly apologies for the extremely belated reply.
Getting to the comments:
1. `TSGForOI` and `MuonTrackingRegionBuilder`: the new Muon L3 code was developed quite actively after I forked to make it run in 2023, it can/should be rebased (my changes are minor).
1. needed Validation changes: what is necessary is in `SimTracker/TrackerHitAssociation/` (this has a significant amount of dependencies, hence it is worth a backport) plus the relevant cfi change to make the muon associator switch to phase2 gemetry in `SimMuon/MCTruth/python/MuonAssociatorByHits_cfi.py`.
1. not needed Validation changes: `HLTFilterToTrackProducer` is just an handy tool and the change in `maxpT` and the additional plots in `MuonTrackValidator.cc` are analysis level changes of the workflow that can be  ignored. 

Said so, @fwyzard, how would you suggest to proceed (e.g. do you want me to get your branch, test the changes and make a PR to your github that you can integrate here)?

**Comment by slava77** [comment]
[This is a weekly check.]
@fwyzard @battibass @gennai 
please clarify on the status of this PR
Based on the last messages here I expect
- updates in this PR to rebase to a more recent 92X version of the muon code
- PR(s) for new necessary code not yet in master

**Comment by fwyzard** [comment]
@battibas, I would suggest to
  - make a PR for any missing functionality for the master/9.4.x release, and make sure everything works there
  - make a PR to backport any missing code to 9.3.x
  - make a PR to backport the same changes to 9.0.x

For the last point, you can start from my branch, but if I remember correctly all the changes came form you in the first place ?

**Comment by gennai** [comment]
9.0.x è un typo spero ...

Best,
S.
Sent from a mobile phone

**Comment by Sam-Harper** [comment]
@battibass  So I find this a bit tiring. I spent some effort in making everything work in 90X for you guys (and like all of us, my free time is limited and many things require my attention) and then you just completely ignored and negated that effort. Do we care about this ever working or not?

**Comment by slava77** [comment]
ping, just as a weekly check

**Comment by slava77** [comment]
-1

to take this off the list of PRs that need imminent attention
updates are expected anyways to proceed

**Comment by Martin-Grunewald** [comment]
-1
superseeded by #20725

**Pull request closed without merging**